rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -468,22 +465,32 @@ trait NodeInfoServiceCached extends NodeInfoService with Loggable with CachedRep
     Full(cache.mapValues(_._2.node))
   }
 
-  def getLDAPNodeInfo(nodeIds: Set[NodeId]): Box[Set[LDAPNodeInfo]] = {
-    if(nodeIds.size > 0) {
-      withUpToDateCache(s"${nodeIds.size} ldap node info") { cache =>
-        Full(cache.collect { case(k, (x,_)) if(nodeIds.contains(k)) => x }.toSet)
-      }
-    } else {
-      Full(Set())
+  override def getLDAPNodeInfo(nodeIds: Set[NodeId], predicats: Seq[NodeInfoMatcher], composition: CriterionComposition): Box[Set[LDAPNodeInfo]] = {
+    def comp(a: Boolean, b: Boolean) = composition match {
+        case And => a && b
+        case Or  => a || b
+    }
+    // combine predicats according to comp
+    def combine(a: NodeInfoMatcher, b: NodeInfoMatcher) = new NodeInfoMatcher {
+      override def matches(node: NodeInfo): Boolean = comp(a.matches(node), b.matches(node))
+    }
+
+    val p = (if(predicats.isEmpty) {
```

Comment:
Maybe a comment to explain how we treat empty case and why it's true for And and false for Or

I understand it now, but will future ourselves will understand it ?